src/features/defi/providers/foxy/components/FoxyManager/Withdraw/components/Approve.tsx

```diff
@@ -55,49 +62,71 @@ export const Approve = ({ onNext }: StepComponentProps) => {
   // user info
   const { state: walletState } = useWallet()
 
-  if (!state || !dispatch) return null
+  const accountFilter = useMemo(() => ({ accountId: accountId ?? '' }), [accountId])
+  const accountMetaData = useAppSelector(state =>
+    selectPortfolioAccountMetadataByAccountId(state, accountFilter),
+  )
 
-  const getWithdrawGasEstimate = async (withdraw: WithdrawValues) => {
-    if (!state.userAddress || !rewardId || !api) return
-    try {
-      const [gasLimit, gasPrice] = await Promise.all([
-        api.estimateWithdrawGas({
-          tokenContractAddress: rewardId,
-          contractAddress,
-          amountDesired: bnOrZero(
-            bn(withdraw.cryptoAmount).times(`1e+${asset.precision}`),
-          ).decimalPlaces(0),
-          userAddress: state.userAddress,
-          type: state.withdraw.withdrawType,
-        }),
-        api.getGasPrice(),
-      ])
-      const returVal = bnOrZero(bn(gasPrice).times(gasLimit)).toFixed(0)
-      return returVal
-    } catch (error) {
-      moduleLogger.error(error, { fn: 'getWithdrawGasEstimate' }, 'getWithdrawGasEstimate error')
-      const fundsError =
-        error instanceof Error && error.message.includes('Not enough funds in reserve')
-      toast({
-        position: 'top-right',
-        description: fundsError
-          ? translate('defi.notEnoughFundsInReserve')
-          : translate('common.somethingWentWrong'),
-        title: translate('common.somethingWentWrong'),
-        status: 'error',
-      })
-    }
-  }
+  const bip44Params = accountMetaData?.bip44Params
 
-  const handleApprove = async () => {
-    if (!rewardId || !state.userAddress || !walletState.wallet || !api) return
+  const getWithdrawGasEstimate = useCallback(
```

Comment:
Mostly just shuffles things around because of order of hooks rule, since this is now a `useCallback`